### PR TITLE
Graceful fail for usage where image isn't found

### PR DESCRIPTION
**Pull request by kenoir**
Performs a get request before attempting an upsert
Fails gracefully when no document found to update
i.e. message is deleted, no error logged

### CODE REVIEW

File: usage/app/controllers/UsageApi.scala [hunk 1]
```diff
@@ -8,7 +8,6 @@ import scala.util.Try
 
 import play.api.libs.json.JsError
 import play.api.Logger
-import play.api.Logger
```
Comment (akash1810) [inline usage/app/controllers/UsageApi.scala]: double import? could remove the additional `import play.api.mvc.Results._` too?

**Comment by akash1810** [comment]
:+1:

File: thrall/app/lib/ElasticSearch.scala [hunk 2]
```diff
@@ -103,6 +103,7 @@ object ElasticSearch extends ElasticSearchClient with ImageFields {
           updateLastModifiedScript,
         scriptType)
       .executeAndLog(s"updating usages on image $id")
+      .recover { case e: DocumentMissingException => {}}
```
Comment (jamesgorrie) [inline thrall/app/lib/ElasticSearch.scala]: We could - to be descriptive of having side effects, have a return of `Future[Unit]` and then match on `UpdateResponse` to return `Unit`.

Or potentially match on `UpdateResponse`and recover on `DocumentMissingException ` to return a new type of `EsActionComplete`?

Comment (jamesgorrie) [inline thrall/app/lib/ElasticSearch.scala]: Completely sidelined - just wondering.

Comment (theefer) [inline thrall/app/lib/ElasticSearch.scala]: +1 to returning unit (`()`).

Comment (theefer) [inline thrall/app/lib/ElasticSearch.scala]: We could log the info here just for sanity of debugging.

**Comment by theefer** [comment]
:+1:

File: thrall/app/lib/ElasticSearch.scala (hunk 2 above)
Comment (kenoir) [inline thrall/app/lib/ElasticSearch.scala]: @theefer `.executeAndLog(s"updating usages on image $id")` does log a descriptive error when it fails already which is why I removed the previous log statement from here as I felt it was redundant.

@jamesgorrie going to release this as is. I'd prefer not to return `Future[Any]` though so maybe you could explain your solution in person when you're around?

File: usage/app/controllers/UsageApi.scala (hunk 1 above)
Comment (kenoir) [inline usage/app/controllers/UsageApi.scala]: Weird. Maybe I just needed to be really really sure.

Have removed.

File: thrall/app/lib/ElasticSearch.scala (hunk 2 above)
Comment (jamesgorrie) [inline thrall/app/lib/ElasticSearch.scala]:
```scala
case class EsActionComplete(action: String)
//...
 def updateImageUsages(id: String, usages: JsValue)(implicit ex: ExecutionContext): Future[EsActionComplete] =
//...
.executeAndLog(s"updating usages on image $id")
.map(_ => EsActionComplete("deleted-image"))
.recover { case e: DocumentMissingException => EsActionComplete("deleted-image-not-found") }
.incrementOnFailure(failedUsagesUpdates) { case e: VersionConflictEngineException => true }
```

**Pull request merged**